rpc/src/v1/impls/eth.rs

```diff
@@ -540,6 +542,23 @@ where
         BlockNumber::Num(n) => BlockId::Number(n),
         BlockNumber::Latest => BlockId::Latest,
         BlockNumber::Earliest => BlockId::Earliest,
+        BlockNumber::Hash {
+            hash,
+            require_canonical,
+        } => {
+            // block check takes precedence over canon check.
+            match client.block_status(BlockId::Hash(hash.clone())) {
+                BlockStatus::InChain => {}
+                _ => return Err(errors::unknown_block()),
+            };
+
+            // TODO Check block hash canonical
```

Comment:
Should be this uncommented?

https://github.com/openethereum/openethereum/pull/10932/files#diff-efc3317730d3b6b64e36557f32d51ffb36c70bf146048c15790d4827fa1f5e12R486-R488